artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -319,6 +321,8 @@
     */
    private final Object directDeliveryGuard = new Object();
 
+   private final CopyOnWriteArrayList<Pair<Consumer, List<MessageReference>>> lingerConsumerRefs = new CopyOnWriteArrayList<>();
```

Comment:
Is this needed surely we can hold the list within the consumer removing need for this altogeather